include/z64transition.h

```diff
@@ -164,31 +165,34 @@ typedef enum {
     /* 86 */ TRANS_TYPE_86 = 86
 } TransitionType;
 
-typedef enum {
-    /* 0 */ FBDEMO_FADE,
-    /* 1 */ FBDEMO_TRIFORCE,
-    /* 2 */ FBDEMO_WIPE1,
-    /* 3 */ FBDEMO_WIPE3,
-    /* 4 */ FBDEMO_WIPE4,
-    /* 5 */ FBDEMO_CIRCLE,
-    /* 6 */ FBDEMO_WIPE5
+#define DEFINE_TRANSITION(enumValue, _structName, _instanceName, _name) enumValue,
+#define DEFINE_TRANSITION_INTERNAL(enumValue, _structName, _instanceName) enumValue,
+
+typedef enum FbDemoType {
+    #include "tables/transition_table.h"
+    /* 7 */ FBDEMO_MAX
 } FbDemoType;
 
+#undef DEFINE_TRANSITION
+#undef DEFINE_TRANSITION_INTERNAL
+
 #define TRANS_NEXT_TYPE_DEFAULT 0xFF
 
+#define DEFINE_TRANSITION(_enumValue, structName, instanceName, _name) structName instanceName;
+#define DEFINE_TRANSITION_INTERNAL(_enumValue, structName, instanceName) structName instanceName;
+
+typedef union TransitionInstance {
+    #include "tables/transition_table.h"
+} TransitionInstance;
```

Comment:
Whoops, I forgot to put a size comment (perhaps just say largest instance struct size?)